src/coalesce.c

```diff
@@ -1,42 +1,56 @@
 #include "data.table.h"
 
-SEXP coalesce(SEXP x, SEXP values, SEXP inplaceArg) {
-
-  if (!isVectorAtomic(x)) error("argument 'x' must be an atomic vector");
+SEXP coalesce(SEXP x, SEXP inplaceArg) {
+  if (TYPEOF(x)!=VECSXP) error("argument 'x' must be a list");
```

Comment:
is this a `nocov`/internal error? since `list(...)` is in R?

Reply:
Oh yes, it is now. Good spot. Will change.